### DISCUSSION TITLE
grafana.ini and persistent user login details

### DISCUSSION THREAD

**Discussion by kappa-wingman** [prometheus-operator/kube-prometheus#1278] [Q&A] [answered]
I had a default installation of kube-prometheus. I had changed default admin password.
I just patched the Deployment/grafana for some environment variables (SMTP and HTTP proxy).
After that, it redeployed by itself, but the admin password is reset to the default.

So my question is:
1) Is there any elegant methods that the existing login password/setting is preserved?
2) Now I need to provide options to grafana.ini for not verifying CA for SMTP TLS setting.
So, I found out that kube-prometheus does not provide a static grafana.ini.
But the promethus operator provide the static config for promethus.yaml by running
$ kubectl get -n monitoring secrets prometheus-k8s -o yaml
It allowed editing of it.
Could a static grafana.ini be configured for this operator?
3) It is not very convenient for editing the jsonnet and rebuild the template every time if there are changes to the SMTP setting.

Thanks in advance.

**Comment by bip7686** [comment, marked as answer]
```
local kp =
  (import 'kube-prometheus/main.libsonnet')
  {
    values+:: {
      common+: {
        namespace: 'monitoring',
      },
      grafana+: {
        config+: {
          sections+: {
            server+: {
              root_url: 'http://grafana.k8s.kube.com',
            },
            security+: {
              admin_password: 'somePassword',
            },
            'auth.anonymous'+: {
              enabled: true,
              org_role: 'Viewer',
            },
            date_formats+: {
              default_timezone: 'browser',
            },
          },
        },
      },

/**** snip *****/
```

I'm configuring grafana like this in my jsonnet file. I believe these settings are saved in the grafana.ini.

**Reply by kappa-wingman** [reply]
The code block looks good, thanks.
